src/core/core.scale.js

```diff
@@ -60,6 +60,111 @@ module.exports = function(Chart) {
+	 * @return object with text placment origin coordinate and alignment
+	 */
+	function useEndOnTruncate(isForward, centerPoint, halfWidth, maxExtent) {
+		var positionProperties = {
+			coord: centerPoint,
+			alignment: 'center'
+		};
+		// Do the start trunction check first to get 'start' alignment when both
+		// ends would truncate
+		if (isForward) {// right side up text
+			if (centerPoint < halfWidth) {// would truncate the start
+				positionProperties.coord = 0;
+				positionProperties.alignment = 'start';
+			} else if (centerPoint + halfWidth > maxExtent) {// would truncate the end
+				positionProperties.coord = maxExtent;
+				positionProperties.alignment = 'end';
+			}// end of (forward) text truncation checks
+		} else if (centerPoint + halfWidth > maxExtent) {// would truncate the start
+			positionProperties.coord = maxExtent;
+			positionProperties.alignment = 'start';
+		} else if (centerPoint < halfWidth) {// would truncate the end
+			positionProperties.coord = 0;
+			positionProperties.alignment = 'end';
+		}// end of (reverse) text truncation checks
+
+		return positionProperties;
+	}// ./useEndOnTruncate(…){
+
+	/**
+	 * Place text, centered if it will fit
+	 *
+	 * Center text, if it fully fits in an area plus margins, otherwise set
+	 *	origin to the outer margin edge, and make that the start/end of the text
+	 *
+	 * Font must be set for the context before calling.	Fails to set (at least)
+	 *	the size if passed in the constraints, and set here.
+	 * Handles horizontal or vertical placment, but not other angles
+	 *
+	 * @param ctx CanvasRenderingContext2D drawing context
+	 * @param text single text string
+	 * @param cnstr object with text placement contraints
+	 * @return object with adjusted text placment properties
+	 */
+	function constrainedTextCenter(ctx, text, cnstr) {
+		var txtProperties;
+		var isForward = cnstr.position !== 'left'; // All text is right side up,
+			// except for Y Axis placed on the left of the chart
+		var halfTextWidth = ctx.measureText(text).width / 2;
+		var out = {
+			positionX: 0,
+			positionY: 0,
+			rotation: 0,
+			hAlign: 'center'
+		};
+
+		if (cnstr.isHorizontal) {// is horizontal text
+			txtProperties = useEndOnTruncate(isForward,
+				midPoint(cnstr.container.left, cnstr.container.right),
+				halfTextWidth,
+				cnstr.container.right + cnstr.container.margins.right);
+			out.positionX = txtProperties.coord;
+
+			out.positionY = cnstr.position === 'bottom' ?
+				cnstr.container.bottom - cnstr.verticalOffset :
+				cnstr.container.top + cnstr.verticalOffset;
+		} else {// not (cnstr.isHorizontal) // is vertical text
+			txtProperties = useEndOnTruncate(isForward,
+				midPoint(cnstr.container.top, cnstr.container.bottom),
+				halfTextWidth,
+				cnstr.container.bottom + cnstr.container.margins.bottom);
+			out.positionY = txtProperties.coord;
+
+			out.positionX = isForward ? // is text on the right side of the chart ?
+				cnstr.container.right - cnstr.verticalOffset :
```

Comment:
Actually, the passed in number is half lineHeight.  This is one of those concept / world view differences.  That value **is** the vertical offset, from the context / view of the text being placed.  The whole reference frame gets rotated, but the text placement itself does not care, and that rotation is done outside of this function.  It does not *know* about that, even though it is doing different calculation based on isHorizontal.  That is an offset from (for) the center line (baseLine) of the text, making it (in a normal context), a vertical offset.  Perhaps `baseLineOffset` will better cover the context for the different reference frames.